[INFO -- 2018-12-01 01:55:02]
Starting flip_value script
version: 1.1
[INFO -- 2018-12-01 01:55:02]
Initialization strings:
[INFO -- 2018-12-01 01:55:02]
(1) file /tmp/knapsack/ga
[INFO -- 2018-12-01 01:55:02]
initSignal:0
[INFO -- 2018-12-01 01:55:02]
endSignal:13
[INFO -- 2018-12-01 01:55:02]
faultModel:0
[EXCEPTION -- 2018-12-01 01:55:03]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[EXCEPTION -- 2018-12-01 01:55:04]
pythonException: empty range for randrange() (0,0, 0)
[EXCEPTION -- 2018-12-01 01:55:05]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[EXCEPTION -- 2018-12-01 01:55:05]
pythonException: empty range for randrange() (0,0, 0)
[EXCEPTION -- 2018-12-01 01:55:05]
pythonException: empty range for randrange() (0,0, 0)
[INFO -- 2018-12-01 01:55:05]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff7490f5d in __GI_abort () at abort.c:90
#2  0x00007ffff74d928d in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7600528 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff74e7650 in malloc_printerr (ar_ptr=0x0, ptr=<optimized out>, str=0x7ffff7600840 "munmap_chunk(): invalid pointer", action=<optimized out>) at malloc.c:5426
#4  munmap_chunk (p=<optimized out>) at malloc.c:2877
#5  __GI___libc_free (mem=<optimized out>) at malloc.c:3138
#6  0x00005555555597ea in __gnu_cxx::new_allocator<int>::deallocate (this=0x7fffffffd800, __p=0x555555783b51) at /usr/include/c++/7/ext/new_allocator.h:125
#7  0x00005555555587a6 in std::allocator_traits<std::allocator<int> >::deallocate (__a=..., __p=0x555555783b51, __n=199) at /usr/include/c++/7/bits/alloc_traits.h:462
#8  0x0000555555557316 in std::_Vector_base<int, std::allocator<int> >::_M_deallocate (this=0x7fffffffd800, __p=0x555555783b51, __n=199) at /usr/include/c++/7/bits/stl_vector.h:180
#9  0x0000555555556b53 in std::_Vector_base<int, std::allocator<int> >::~_Vector_base (this=0x7fffffffd800, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:162
#10 0x0000555555555e7d in std::vector<int, std::allocator<int> >::~vector (this=0x7fffffffd800, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:435
#11 0x0000555555555329 in crossover (pai=std::vector of length 200, capacity 200 = {...}, mae=std::vector of length 200, capacity 200 = {...}, filho1=std::vector of length 200, capacity 200 = {...}, filho2=std::vector of length 200, capacity 200 = {...}) at knapsackga.cpp:72
#12 0x00005555555555c7 in genetic_algorithm () at knapsackga.cpp:104
#13 0x0000555555555a9b in main () at knapsackga.cpp:141

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Memory content before bitflip:0xc80x000x000x00
Fault Model: Single bit-flip
base address to flip value: 0x55555575f178 
address max offset: 4
Memory content after  bitflip:0x880x000x000x00
frame name: std::allocator_traits<std::allocator<int> >::deallocate
symbol name: n
symbol filename: knapsackga.cpp
symbol line: 16
value: 136
value address: 0x55555575f178 <n>
Type: The type is an integer type.
Type sizeof: 4

[INFO -- 2018-12-01 01:55:05]
Fault Injection Successful after 3.45265793800354s
[INFO -- 2018-12-01 01:55:05]
event type: exit
[EXCEPTION -- 2018-12-01 01:55:05]
exit code: no exit code available
